### PR TITLE
feat(er/attachment): support new data source

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
New data source is support: huaweicloud_er_attachments
This data source can query all kinds of ER attachments (VPN, VGW, PEERING, VPC) during follow arguments:
- attachment_id
- type
- status
- name
- tags

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new data source to query ER attachment.
2. support related documentation and acc tests.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccAttachmentsDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccAttachmentsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccAttachmentsDataSource_basic
=== PAUSE TestAccAttachmentsDataSource_basic
=== CONT  TestAccAttachmentsDataSource_basic
--- PASS: TestAccAttachmentsDataSource_basic (90.08s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        90.157s
```
```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccAttachmentsDataSource_filterById'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccAttachmentsDataSource_filterById -timeout 360m -parallel 4
=== RUN   TestAccAttachmentsDataSource_filterById
=== PAUSE TestAccAttachmentsDataSource_filterById
=== CONT  TestAccAttachmentsDataSource_filterById
--- PASS: TestAccAttachmentsDataSource_filterById (91.45s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        91.538s
```
```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccAttachmentsDataSource_filterByType'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccAttachmentsDataSource_filterByType -timeout 360m -parallel 4
=== RUN   TestAccAttachmentsDataSource_filterByType
=== PAUSE TestAccAttachmentsDataSource_filterByType
=== CONT  TestAccAttachmentsDataSource_filterByType
--- PASS: TestAccAttachmentsDataSource_filterByType (98.01s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        98.175s
```
```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccAttachmentsDataSource_filterByStatus'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccAttachmentsDataSource_filterByStatus -timeout 360m -parallel 4
=== RUN   TestAccAttachmentsDataSource_filterByStatus
=== PAUSE TestAccAttachmentsDataSource_filterByStatus
=== CONT  TestAccAttachmentsDataSource_filterByStatus
--- PASS: TestAccAttachmentsDataSource_filterByStatus (99.21s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        99.348s
```
```
make testacc TEST='./huaweicloud/services/acceptance/er' TESTARGS='-run=TestAccAttachmentsDataSource_filterByTags'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/er -v -run=TestAccAttachmentsDataSource_filterByTags -timeout 360m -parallel 4
=== RUN   TestAccAttachmentsDataSource_filterByTags
=== PAUSE TestAccAttachmentsDataSource_filterByTags
=== CONT  TestAccAttachmentsDataSource_filterByTags
--- PASS: TestAccAttachmentsDataSource_filterByTags (97.48s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/er        97.580s
```

### CODE REVIEW

File: huaweicloud/services/er/data_source_huaweicloud_er_attachments.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	if attachmentId, ok := d.GetOk("attachment_id"); ok {
+		filter["ID"] = attachmentId
+	}
+
+	if len(filter) < 1 {
+		return all, nil
+	}
+	filterResult, err := utils.FilterSliceWithField(all, filter)
+	if err != nil {
+		return nil, fmt.Errorf("error filting attachment list: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/er/data_source_huaweicloud_er_attachments.go]: filtering

File: huaweicloud/services/er/data_source_huaweicloud_er_attachments.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	}
+	result := make([]attachments.Attachment, 0, len(filterResult))
+	for _, val := range filterResult {
+		result = append(result, val.(attachments.Attachment))
+	}
```
Comment (deer-hang) [inline huaweicloud/services/er/data_source_huaweicloud_er_attachments.go:139]:
```suggestion
	result := make([]attachments.Attachment, len(filterResult))
	for index, val := range filterResult {
		result[index] = val.(attachments.Attachment)
	}
```

Comment (Lance52259) [reply huaweicloud/services/er/data_source_huaweicloud_er_attachments.go:139]: They are not different

File: huaweicloud/services/er/data_source_huaweicloud_er_attachments.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/er/data_source_huaweicloud_er_attachments.go]: fixed.

File: docs/data-sources/er_attachments.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+# huaweicloud_er_attachments
+
+Use this data source to filter ER attachments within Huaweicloud.
+
+## Example Usage
+
+```hcl
+data "huaweicloud_er_attachments" "test" {
+  tags = {
```
Comment (ShiChangkuo) [inline docs/data-sources/er_attachments.md:17]: Missing required parameter in this usage

Comment (Lance52259) [reply docs/data-sources/er_attachments.md:17]: fixed.

**Pull request merged**